### PR TITLE
docs: Add swagger and readme documentation

### PR DESCRIPTION
**Pull request by domachine**
This PR adds a readme and a swagger resource.

### CODE REVIEW

**Review by valentindey — APPROVED** [review]
LGTM - I skipped checking the generated docs

File: README.md [hunk 1]
```diff
@@ -1 +1,75 @@
...
+
+## About The Project
+
+This is a service to validate documents against the CSAF standard. It uses the csaf-validator-lib under the hood which included as a git subtree module.
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
+## Documentation
+
+The documentation is available as a swagger resource provided by the service itself under `/docs`. So once the server is running, visit http://localhost:3000/docs in your browser.
```
Comment (tschmidtb51) [inline README.md:34]: > http://localhost:3000/docs

Is that fix or definable/dependent by variables?

File: README.md [hunk 2]
```diff
@@ -1 +1,75 @@
...
+      <a href="#development">Development</a>
+      <ul>
+        <li><a href="#prerequisites">Prerequisites</a></li>
+        <li><a href="#installation">Installation</a></li>
+        <li><a href="#run-server">Run server</a></li>
+      </ul>
+    </li>
+    <li><a href="#testing">Testing</a></li>
+  </ol>
+</details>
```
Comment (tschmidtb51) [inline README.md:24]: Same here:
```suggestion
<!-- TOC depthfrom:2 depthto:3 -->

- [About the Project](#about-the-project)
- [Documentation](#documentation)
- [...]

<!-- /TOC -->

File: README.md [hunk 3]
```diff
@@ -1 +1,75 @@
...
+
+## Documentation
+
+The documentation is available as a swagger resource provided by the service itself under `/docs`. So once the server is running, visit http://localhost:3000/docs in your browser.
+
+## Development
+
+### Prerequisites
+
+You need at least Node.js version 14 or higher.
```
Comment (tschmidtb51) [inline README.md:40]:
```suggestion
You need at least **Node.js version 14 or higher**. [Nodesource](https://github.com/nodesource/distributions/blob/master/README.md) provides binary distributions for various Linux distributions.

File: backend/lib/app.js [hunk 4]
```diff
@@ -2,6 +2,22 @@
  * @param {import('fastify').FastifyInstance} fastify
  */
 export default async function (fastify) {
+  fastify.register(import('fastify-swagger'), {
+    routePrefix: '/docs',
+    openapi: {
+      info: {
+        title: 'CSAF Validator Service',
+        version: '1.0.0',
```
Comment (tschmidtb51) [inline backend/lib/app.js:10]: Isn't that the version of the server? If so shouldn't that be retrieved from a Git tag?
(Maybe that's something for later - if so, please add as issue and ignore in the first round :-))

File: documents/generated/asciidoc/index.adoc [hunk 5]
```diff
@@ -0,0 +1,378 @@
+= CSAF Validator Service
+team@openapitools.org
```
Comment (tschmidtb51) [inline documents/generated/asciidoc/index.adoc:2]: This looks like a default... Should we override that? Or is it from the generator?
If so,I  guess that should be the address of the project... e.g. secvisogram[at]bsi.bund.de

File: documents/generated/asciidoc/index.adoc [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+1.0.0
+:toc: left
+:numbered:
+:toclevels: 3
+:source-highlighter: highlightjs
+:keywords: openapi, rest, CSAF Validator Service
+:specDir: 
+:snippetDir: 
+:generator-template: v1 2019-12-20
+:info-url: https://openapi-generator.tech
```
Comment (tschmidtb51) [inline documents/generated/asciidoc/index.adoc:12]: Same here.
If it should be project specific, please use the Github repo...

File: documents/generated/asciidoc/index.adoc [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+`GET /api/v1/tests`
+
+Retrieve all tests.
+
+===== Description
+
+This endpoint is intended to be used to discover all available tests.
+
+
+// markup not found, no include::{specDir}api/v1/tests/GET/spec.adoc[opts=optional]
```
Comment (tschmidtb51) [inline documents/generated/asciidoc/index.adoc:43]: Is something missing here?

**Review by tschmidtb51 — COMMENTED** [review]
Looks like a solid start. Have look at my comment whether you want to address them directly or as separate issues/PRs. I'm fine with both...

File: README.md (hunk 1 above)
Comment (domachine) [reply README.md:34]: This is definable with using configuration files in `config/`. I'll note this here.

File: README.md [hunk 8]
```diff
@@ -1 +1,75 @@
-# csaf-validator-service
\ No newline at end of file
+<div id="top"></div>
+
+<br />
+<div align="center">
+<h3 align="center">csaf-validator-service</h3>
+</div>
```
[the remark this answers is not in the document]
Comment (domachine) [reply README.md:6]: I used the same style in the `csaf-validator-lib` but it's a good point, to find a consistent style here.

File: README.md (hunk 2 above)
Comment (domachine) [reply README.md:24]: Agreed

File: backend/lib/app.js (hunk 4 above)
Comment (domachine) [reply backend/lib/app.js:10]: Yes, this might be an option. I'll add this as an issue.

File: documents/generated/asciidoc/index.adoc (hunk 5 above)
Comment (domachine) [reply documents/generated/asciidoc/index.adoc:2]: Yes, this should probably be a specific credits part. I'll insert the secvisogram[at]bsi.bund.de email here if that is okay.

File: documents/generated/asciidoc/index.adoc (hunk 7 above)
Comment (domachine) [reply documents/generated/asciidoc/index.adoc:43]: These are optional samples that I left out for the moment.

File: backend/lib/app.js (hunk 4 above)
Comment (tschmidtb51) [reply backend/lib/app.js:10]: Added as issue: https://github.com/secvisogram/csaf-validator-service/issues/5#issue-1196871889

File: README.md [hunk 9]
```diff
@@ -1 +1,71 @@
-# csaf-validator-service
\ No newline at end of file
+<div id="top"></div>
+
+# CSAF Validator Service
+
+<!-- toc -->
```
Comment (tschmidtb51) [inline README.md:5]: The pre-commit script to update the TOC: Does it also run locally or just on Github?

File: .husky/pre-commit [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run process-readme
```
Comment (tschmidtb51) [inline .husky/pre-commit:4]: What is the husky for?

File: documents/generated/html/index.html [hunk 11]
```diff
@@ -0,0 +1,423 @@
...
+  </head>
+  <body>
+  <h1>CSAF Validator Service</h1>
+    <div class="app-desc">This is a service to validate documents against the CSAF standard.</div>
+    <div class="app-desc">More information: <a href="https://github.com/secvisogram/csaf-validator-service">https://github.com/secvisogram/csaf-validator-service</a></div>
+    <div class="app-desc">Contact Info: <a href="secvisogram@bsi.bund.de">secvisogram@bsi.bund.de</a></div>
+    <div class="app-desc">Version: 1.0.0</div>
+    <div class="app-desc">BasePath:</div>
+    <div class="license-info">All rights reserved</div>
+    <div class="license-url">http://apache.org/licenses/LICENSE-2.0.html</div>
```
Comment (tschmidtb51) [inline documents/generated/html/index.html:189]: Why is Apache 2.0 listed here? It should be MIT...

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Found a licensing issue :see_no_evil:

File: README.md (hunk 9 above)
Comment (domachine) [reply README.md:5]: It runs locally only to update the readme toc.

File: .husky/pre-commit (hunk 10 above)
Comment (domachine) [reply .husky/pre-commit:4]: This provides the git hook to process the readme toc right before committing.

File: documents/generated/html/index.html (hunk 11 above)
Comment (domachine) [reply documents/generated/html/index.html:189]: You're absolutely right! I just missed to fill this value in and Apache seems to be the default. I'm still finding my way into swagger 🙈 .

File: backend/lib/app.js [hunk 12]
```diff
@@ -2,6 +2,27 @@
...
+        description:
+          'This is a service to validate documents against the CSAF standard.',
+        contact: {
+          url: 'https://github.com/secvisogram/csaf-validator-service',
+          email: 'secvisogram@bsi.bund.de',
+        },
+        license: {
+          name: 'MIT',
+          url: 'https://spdx.org/licenses/MIT.html',
```
Comment (tschmidtb51) [inline backend/lib/app.js:19]: Is that correct that we link to the SPDX site or should we use https://github.com/secvisogram/csaf-validator-service/blob/main/LICENSE ? (Just a question - no preference or idea how it is used correctly.)

Comment (domachine) [reply backend/lib/app.js:19]: Yes, probably it's better to refer to the project's license file.

**Review by tschmidtb51 — APPROVED** [review]
Thank you. LTGM now :smile:

**Pull request merged**